test/unit/utils/dom-data.test.js

```diff
@@ -21,3 +22,45 @@ QUnit.test('should get and remove data from an element', function(assert) {
 
   assert.notOk(DomData.hasData(el), 'cached item emptied');
 });
+
+let memoryTestRun = false;
+
+QUnit.done(function(details) {
+  // don't run the extra dom data test on failures, there will likely be
+  // memory leaks
+  if (details.failed || memoryTestRun) {
+    return;
+  }
+
+  // TODO: fix memory leaks on the following
+  if (videojs.browser.IS_SAFARI || videojs.browser.IS_EDGE || videojs.browser.IE_VERSION) {
+    return;
+  }
+
+  memoryTestRun = true;
+
+  QUnit.module('dom-data memory');
+
+  /**
+   * If this test fails you will want to add a debug statement
+   * in DomData.getData with the `id`. For instance if DomData.elData
+   * had 2 objects in it {5: {...}, 2003: {...} you would add:
+   *
+   * ```js
+   * if (id === 5) {
+   *   debugger;
+   * }
+   * ```
+   * to the tests to see what test. Then re-run the tests to see
+   * what leaking and where.
+   *
+   * > Note that the id can be off by 1-2 in either direction
+   *   for larger guids, so you may have to account for that.
+   */
+  QUnit.test('Memory is not leaking', function(assert) {
+    if (Object.keys(DomData.elData).length > 0) {
+      videojs.domData = DomData;
```

Comment:
This is here for easier debugging for when/if this test fails?

Reply:
yeah